dax/mds/client/client.go

```diff
@@ -159,7 +159,7 @@ func (c *Client) DatabaseByName(ctx context.Context, orgID dax.OrganizationID, n
 
 	if resp.StatusCode != http.StatusOK {
 		b, _ := io.ReadAll(resp.Body)
-		return nil, errors.Errorf("status code: %d: %s", resp.StatusCode, b)
+		return nil, errors.Wrapf(errors.UnmarshalJSON(b), "status code: %d", resp.StatusCode)
```

Comment:
why don't we have errors.UnmarshalJSON take a reader and wrap the ReadAll up in there?

Reply:
that's a good idea. I'll make that change and then convert the rest of the methods in our service apis to use this.